account_banking_mandate/model/account_banking_mandate.py

```
@@ -0,0 +1,185 @@
+                raise orm.except_orm(
+                    _('Error:'),
+                    _("The mandate '%s' can't have a date of last debit "
+                        "before the date of signature.")
+                    % mandate.unique_mandate_reference)
+        return True
+
+    def _check_valid_state(self, cr, uid, ids):
+        for mandate in self.browse(cr, uid, ids):
+            if mandate.state == 'valid' and not mandate.signature_date:
+                raise orm.except_orm(
+                    _('Error:'),
+                    _("Cannot validate the mandate '%s' without a date of "
+                        "signature.")
+                    % mandate.unique_mandate_reference)
+            if mandate.state == 'valid' and not mandate.partner_bank_id:
+                raise orm.except_orm(
+                    _('Error:'),
+                    _("Cannot validate the mandate '%s' because it is not "
+                        "attached to a bank account.")
+                    % mandate.unique_mandate_reference)
+        return True
+
+    _constraints = [
+        (_check_dates, "Error msg in raise",
+            ['signature_date', 'last_debit_date']),
+        (_check_valid_state, "Error msg in raise",
+            ['state', 'partner_bank_id']),
+    ]
+
+    def mandate_partner_bank_change(
+            self, cr, uid, ids, partner_bank_id, last_debit_date, state):
+        res = {'value': {}}
+        if partner_bank_id:
+            partner_bank_read = self.pool['res.partner.bank'].read(
+                cr, uid, partner_bank_id, ['partner_id'])['partner_id']
+            if partner_bank_read:
+                res['value']['partner_id'] = partner_bank_read[0]
+        return res
+
+    def validate(self, cr, uid, ids, context=None):
+        to_validate_ids = []
+        for mandate in self.browse(cr, uid, ids, context=context):
+            assert mandate.state == 'draft', 'Mandate should be in draft state'

```

Not blocking but I would return an except_orm Exception, it will be more user friendly